app/controllers/comments_controller.rb

```diff
@@ -0,0 +1,50 @@
+class CommentsController < ApplicationController
+
+  def index
+    @comments = Comment.all.order(:id).reverse
+    @post = @comment.post
+  end
+
+  def new
+    @post = Post.find(params[:post_id])
+    @comment = Comment.new
+  end
+
+  def create
+    @post = Post.find(params[:post_id])
+    @comment = @post.comments.create( comment_params )
```

Comment:
Instead of letting the user POST a user value, you can use the logged in user:

 ```rb
@post.comments.create( comment_params.merge(user: session[:user]) )
```